CONTRIBUTING.md

```diff
@@ -42,7 +42,95 @@ you have written a test like this, make sure it does not fail often by
 uncommenting the code that sets a random seed in `tests/main.cpp` and running
 your test many times.
 
-Information on how to build and run the tests is in the main README.md file.
+### Install Dependencies
+
+<details open>
+<summary>Linux</summary>
+
+Use your distributions' package manager to install the required dependencies. The commands shown below are for Ubuntu.
+
+```bash
+$ sudo apt-get update
+$ sudo apt-get install liblapack-dev
+$ sudo apt-get install libarmadillo-dev
+```
+</details>
+
+<details open>
+<summary>MacOS</summary>
+
+Use a package manager like [Homebrew](https://brew.sh) to get the necessary dependencies.
+
+````bash
+$ brew install lapack armadillo
+````
+</details>
+
+<details open>
+<summary>Windows</summary>
+
+You can install **ensmallen** directly by using [vcpkg](https://github.com/microsoft/vcpkg).
+
+```
+vcpkg install ensmallen:x64-windows
+```
+</details>
+
+### Build and Test
+
+This section describes how to build and test the **ensmallen** library from source. **ensmallen** uses CMake as its build system.
+
+First, clone the source code from Github and change into the cloned directory. Or alternatively, you can download the latest relese from the [website](http://ensmallen.org) and extract it.
+
+```bash
+$ git clone https://github.com/mlpack/ensmallen
+$ cd ensmallen
+
+# - or -
+
+$ wget http://ensmallen.org/files/ensmallen-latest.tar.gz
+$ tar -xvzpf ensmallen-latest.tar.gz
+$ cd ensmallen-latest
```

Comment:
WE packaged the latest version, so right now the folder would be `ensmallen-2.10.5` and not `ensmallen-latest`, so I think we can use `ensmallen-x.x.x`.

Reply:
Will I keep the semver as x.x.x, or should we bump it manually for every release?

Reply:
We could use the actual version here, we just have too update the release script: https://github.com/mlpack/ensmallen/blob/master/rel/ensmallen-release.sh accordingly.

Reply:
Okay, on it!

Reply:
Great, thanks!